src/app/core/services/notifications/notification-wrapper.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@angular/core'
+
+import { DefaultNotificationType } from '../../../../assets/data/defaultConfig'
+import { ConfigKeys } from '../../../shared/enums/config'
+import { NotificationMessagingType } from '../../../shared/models/notification-handler'
+import { RemoteConfigService } from '../config/remote-config.service'
+import { AppServerRestNotificationService } from './app-server-rest-notification.service'
+import { FcmXmppNotificationService } from './fcm-xmpp-notification.service'
+import { LocalNotificationService } from './local-notification.service'
+import { NotificationService } from './notification.service'
+
+@Injectable()
+export class NotificationWrapperService extends NotificationService {
+  notificationService: NotificationService
+
+  constructor(
+    public appServerRestNotificationService: AppServerRestNotificationService,
+    public fcmXmppNotificationService: FcmXmppNotificationService,
+    public localNotificationService: LocalNotificationService,
+    private remoteConfig: RemoteConfigService
+  ) {
+    super()
+  }
+
+  init() {
+    return this.remoteConfig
+      .read()
+      .then(config =>
+        config.getOrDefault(
+          ConfigKeys.NOTIFICATION_MESSAGING_TYPE,
+          DefaultNotificationType
+        )
+      )
+      .then(type => {
+        if (type == NotificationMessagingType.LOCAL)
+          return (this.notificationService = this.localNotificationService)
+        if (type == NotificationMessagingType.FCM_REST)
+          return (this.notificationService = this.appServerRestNotificationService)
+        return (this.notificationService = this.fcmXmppNotificationService)
```

Comment:
Use a `switch` statement here and throw an Error in the default block stating `no such notification service available.`. Also, add a case for `FCM_XMPP`.